File: Maximizar.js
/*
En este código, la función maximizeNumber convierte el número en una cadena, luego lo divide en un array de caracteres, 
lo ordena en orden descendente y luego lo une nuevamente para formar el número máximo posible. 
Esto se logra al tener los dígitos más grandes en las posiciones más significativas.
*/
function maximizeNumber(num) {
    const numStr = num.toString();
    const numArray = numStr.split('');
    numArray.sort((a, b) => b - a);
    const maxNum = parseInt(numArray.join(''));
    return maxNum;
}

// Ejemplo de uso
const valor = 120;
const maximizedValue = maximizeNumber(valor);
console.log(`El número más grande posible al reorganizar los dígitos de ${valor} es ${maximizedValue}.`);

File: Maximizar.py
"""
En este programa, la función maximize_number convierte el número en una cadena, luego crea una lista de caracteres a partir de la cadena, 
la ordena en orden descendente y luego une los caracteres nuevamente para formar el número máximo posible. 
Esto se logra al tener los dígitos más grandes en las posiciones más significativas.
"""
def maximize_number(num):
    num_str = str(num)
    num_list = list(num_str)
    num_list.sort(reverse=True)
    max_num = int(''.join(num_list))
    return max_num

# Ejemplo de uso
valor = 120
maximized_value = maximize_number(valor)
print(f"El número más grande posible al reorganizar los dígitos de {valor} es {maximized_value}.")
